kedro/contrib/io/dask/dask_parquet.py

```diff
@@ -0,0 +1,131 @@
+# NONINFRINGEMENT. IN NO EVENT WILL THE LICENSOR OR OTHER CONTRIBUTORS
+# BE LIABLE FOR ANY CLAIM, DAMAGES, OR OTHER LIABILITY, WHETHER IN AN
+# ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF, OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# The QuantumBlack Visual Analytics Limited ("QuantumBlack") name and logo
+# (either separately or in combination, "QuantumBlack Trademarks") are
+# trademarks of QuantumBlack. The License does not grant you any right or
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+#     or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""``ParquetDaskDataSet`` is a data set used to load and save data to parquet files using Dask"""
+from copy import deepcopy
+from typing import Any, Dict
+
+import dask.dataframe as dd
+import fsspec
+from fsspec.utils import infer_storage_options
+
+from kedro.contrib.io import DefaultArgumentsMixIn
+from kedro.io.core import AbstractDataSet
+
+_PROTOCOL_DELIM = "://"
+
+
+class ParquetDaskDataSet(DefaultArgumentsMixIn, AbstractDataSet):
+    """``ParquetDaskDataSet`` loads and saves data to parquet file(s). It uses Dask
+    remote data services to handle the corresponding load and save operations:
+        https://docs.dask.org/en/latest/remote-data-services.html
+
+        Example (AWS S3):
+        ::
+
+            >>> from kedro.contrib.io.dask.dask_parquet import ParquetDaskDataSet
+            >>> import pandas as pd
+            >>> import dask.dataframe as dd
+            >>>
+            >>> data = pd.DataFrame({'col1': [1, 2], 'col2': [4, 5],
+            >>>                      'col3': [5, 6]})
+            >>> ddf = dd.from_pandas(data, npartitions=2)
+            >>>
+            >>> data_set = ParquetDaskDataSet(
+            >>>                         filepath="s3://bucket_name/path/to/folder",
+            >>>                         storage_options={
+            >>>                             'client_kwargs':
+            >>>                                 'aws_access_key_id': 'YOUR_KEY',
+            >>>                                 'aws_secret_access_key': 'YOUR SECRET'},
+            >>>                         save_args={"compression": "GZIP"})
+            >>> data_set.save(ddf)
+            >>> reloaded = data_set.load()
+            >>>
+            >>> assert ddf.compute().equals(reloaded.compute())
+    """
+
+    DEFAULT_SAVE_ARGS = {"write_index": False}
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        filepath: str,
+        storage_options: Dict[str, Any] = None,
```

Comment:
Maybe combine `storage_options` with `credentials` similar to how @andrii-ivaniuk did it in `CSVDataSet` with `credentials` and `fs_args`.

Reply:
`storage_options` already covers both `fs_args` and `credentials`. The difference from `CSVDataSet` is that we never instantiate `fsspec` instance here. It is instantiated inside dask dd API when save or load is called.